### ISSUE TITLE
Transfer token error

### ISSUE THREAD

**Issue by vsdfnvuitsb** [michaelhly/solana-py#59]
I'm trying to transfer token,this's my code:
`import base58
from solana.publickey import PublicKey
from solana.account import Account
from solana.rpc.api import Client
from solana.transaction import Transaction
from spl.token.instructions import transfer, TransferParams

# acc = Account(bytes(prk[:32]))

testtoken = PublicKey("6YdsfSpcU5RpU3uFWsPNGmfHPyxELPCqQu2hg5deYtDv")
sender = Account(bytes(prk[:32])) 
dest = 'G8Sop5bHcEkK52mnad6M5PitijWH2oRBxSFzz5bMDK9'
transfer_params = TransferParams(
     amount=3,
     dest=PublicKey(base58.b58decode(dest)),
     owner=sender.public_key(),
     program_id=testtoken,
     source=sender.public_key())
txn = Transaction()
txn.add(transfer(transfer_params))
solana_client = Client("https://testnet.solana.com")
solana_client.send_transaction(txn, sender)`

But response:
`{'code': -32002, 'message': 'Transaction simulation failed: This program may not be used for executing instructions', 'data': {'err': 'InvalidProgramForExecution', 'logs': []}}
`
`Exception: Failed to send transaction`

**Comment by michaelhly** [comment]
program_id should not be `testtoken`, it should be https://michaelhly.github.io/solana-py/spl.html#spl.token.constants.TOKEN_PROGRAM_ID

**Comment by vsdfnvuitsb** [comment]
> program_id should not be `testtoken`, it should be https://michaelhly.github.io/solana-py/spl.html#spl.token.constants.TOKEN_PROGRAM_ID

After I changed the token address to the main network address, I still made the same error
![image](https://user-images.githubusercontent.com/70013651/117388715-ce995600-af1d-11eb-9301-ab3fd7caaad3.png)
`{'code': -32002, 'message': 'Transaction simulation failed: This program may not be used for executing instructions', 'data': {'err': 'InvalidProgramForExecution', 'logs': []}}
`
`Exception: Failed to send transaction`

**Comment by michaelhly** [comment]
No. You need to provide `spl.token.constants.TOKEN_PROGRAM_ID` as `programID`.

**Comment by vsdfnvuitsb** [comment]
> No. You need to provide `spl.token.constants.TOKEN_PROGRAM_ID` as `programID`.

Where can I see it

**Comment by michaelhly** [comment]
`TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA`

**Comment by ShimTeacher** [comment]
@Mitolius hey guys im same problem 
Shall we solve the problem together?

**Comment by geoffwhittington** [comment]
I hit the same issue and resolved it by ensuring that the `source` and `dest` values were token account addresses.

**Comment by MarioProjects** [comment]
> I hit the same issue and resolved it by ensuring that the `source` and `dest` values were token account addresses.

I did, the send_transaction worked and got an TX, but no token was sent and the TX displays nothing. 

I am trying to send a custom SPL token I want to distribute to my holders. This is my code:

```python
from solana.publickey import PublicKey
from solana.account import Account
from solana.rpc.api import Client
from solana.transaction import Transaction
from solana.keypair import Keypair
from spl.token.instructions import transfer, TransferParams
from spl.token.constants import TOKEN_PROGRAM_ID

client = Client("https://api.mainnet-beta.solana.com")

secret_key= [...]
    
sender = Keypair.from_secret_key(bytes(secret_key))
receiver = PublicKey("BkoXyZy4Poam3E97Urd7rmwAvcyC3tZFBap1GHvFCjdS") # TOKEN Account

amount = int(10e6 * 5)

chick_coin = PublicKey("3RXTenk2bvx5cTYY8JrN94EFi3AkYCz9tzJzE7oRhSgf") # Mainnet address
chick_acc = PublicKey("6u4vEaj59irczDu17ydB6E9DTM4XVXZErTVe9LvFyNUG")

# https://michaelhly.github.io/solana-py/spl/intro/
transaction = Transaction().add(transfer(TransferParams(
    source=chick_acc, # PK of token account you're transfering FROM
    dest=receiver, # PK of token account you're transfering TO
    owner=sender.public_key, # Account that owns your source token account
    program_id = TOKEN_PROGRAM_ID,
    amount=amount
)))

res = client.send_transaction(transaction, sender)
```

Obtaining next result:
```
{
  'jsonrpc': '2.0',
  'result': '9cmfZ1zHkxwwUMzHdNAc15Ld2y5Y1PDbRpeuPhLcniPAEFRPTQrHW2akt3cpiYUX5Sgi2rSgTVz3zQPk99jaf2h',
  'id': 2
}
```

But the tokens sometimes are transferred, most of the time no, and never got a bad/error result from the res...

Can you help me? @vcyleung @michaelhly
